oauthproxy.go

```diff
@@ -838,6 +839,27 @@ func (p *OAuthProxy) enrichSessionState(ctx context.Context, s *sessionsapi.Sess
 func (p *OAuthProxy) AuthOnly(rw http.ResponseWriter, req *http.Request) {
 	session, err := p.getAuthenticatedSession(rw, req)
 	if err != nil {
+		if authOnlyRedirectSignIn(req) {
```

Comment:
Maybe flip this to:

```suggestion
		if !authOnlyRedirectSignIn(req) {
				http.Error(rw, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
				return
		}
```

So we can eliminate the deep nesting for all that custom logic below

Reply:
updated